homeassistant/config_entries.py

```diff
@@ -529,9 +528,6 @@ async def async_migrate(self, hass: HomeAssistant) -> bool:
                 "Flow handler not found for entry %s for %s", self.title, self.domain
             )
             return False
-        # Handler may be a partial
-        while isinstance(handler, functools.partial):
```

Comment:
Why don't we need this now?

Reply:
AFAICT this isn't used anymore. Except for `sonos` all handlers are registered through `__init_subclass__`. (I'll open a PR to fix that in `sonos` next.)
https://github.com/home-assistant/core/blob/c59115bb4f571f1f618693f9f060d63fcd2b3302/homeassistant/config_entries.py#L1188-L1192

Thus `HANDLERS` can be defined as `Registry[str, type[ConfigFlow]]` which resulted in `mypy` identifying the body of the while loop as unreachable.

Furthermore, the while loop isn't covered by tests.

--
Did I miss something?

Reply:
I think we need to keep this for legacy compatibility with custom integrations.

Reply:
https://github.com/home-assistant/core/pull/67087/commits/7fc23d80b5851ab77e0c66634cf84550f996227b